group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupCoordinatorServiceTest.java

```diff
@@ -279,6 +291,162 @@ public void testConsumerGroupHeartbeatWithException(
         );
     }
 
+    @Test
+    public void testConsumerHeartbeatRequestValidation() throws ExecutionException, InterruptedException, TimeoutException {
```

Comment:
This is a copy of the original tests but I had to adapt them to work with futures.